userspace/libsinsp/filter.cpp

```diff
@@ -272,14 +247,12 @@ sinsp_filter* sinsp_filter_compiler::compile()
 	}
 	catch (const sinsp_exception& e)
 	{
-		delete new_filter;
-		m_filter = NULL;
+		m_filter = nullptr;
 		throw e;
 	}
 
 	// return compiled filter
-	m_filter = NULL;
-	return new_filter;
+	return std::move(m_filter);
```

Comment:
Aren't we leaving an invalid object as part of the current `sinsp_filter_compiler` by moving a member? This seems dangerous.

Reply:
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3690.pdf

From the standard, it is my understanding that a unique pointer gets nullified after getting moved.

> Additionally, u can, upon request, transfer ownership to another unique pointer u2. Upon completion of such a transfer, the following postconditions hold:
> — u2.p is equal to the pre-transfer u.p,
> — u.p is equal to nullptr, and
> — if the pre-transfer u.d maintained state, such state has been transferred to u2.d.